### DISCUSSION TITLE
Add back option to disable app router

### DISCUSSION THREAD

**Discussion by RDIL** [vercel/next.js#55671] [Ideas]
### Goals

Add an option to disable the app router.

### Non-Goals

Remove the app router.

### Background

Over the last 24 hours, I've been doing extensive testing and here are my findings:

Note: all this testing is done on a site which **only** uses pages.

Next 13.4 with appDir unset or set to true:

- Average compile time (hot reload): ~700ms
- Average memory usage after running `yarn next start` (production build) for a couple minutes: 600mb

Next 13.4 with appDir *explicitly set to false*:

- Average compile time (hot reload): ~400ms
- Average memory usage after running `yarn next start` (production build) for a couple minutes: 200mb

Next 13.5:

- Average compile time (hot reload): ~800ms
- Average memory usage after running `yarn next start` (production build) for a couple minutes: 400mb

All of this testing was done on my M1 Mac laptop.
To say the least, these numbers are very disappointing. This blocks me from upgrading Next in my apps until there is a way to stop the app directory from being loaded.
I don't know what exactly is happening inside Next but I can say that the app directory being enabled has very much impacted performance.

### Proposal

Revert #54785

**Comment by aka-bassman** [comment]
Same opinion. Need to control the enable/disalbe of appDir.
I need to create both app & pages dir to make both SSR(app) and CSR(pages) services.
now, I downgraded version 13.4.19 now.
